src/app.jsx

```diff
@@ -55,38 +56,44 @@ export const Application = () => {
     useEffect(() => {
         if (path === undefined)
             return;
-        setFiles([]);
-        const currentPath = path.slice(0, pathIndex).join("/");
-        const channel = cockpit.channel({
-            payload: "fslist1",
-            path: `/${currentPath}`,
-            superuser: "try",
-            watch: true,
-        });
 
-        channel.addEventListener("message", (ev, data) => {
-            const item = JSON.parse(data);
-            if (item.event === 'present' && item.path[0] !== ".")
-                setFiles(f => [...f, { name: item.path, type: item.type, modified: item.modified }]);
-            else if (item.event === 'deleted') {
-                const deleted_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                setFiles(f => f.filter((res) => { return res.name !== deleted_name }));
-            } else if (item.event === 'created') {
-                const created_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                const date_modified = item.tag.slice(item.tag.indexOf("-") + 1, item.tag.lastIndexOf("."));
-                if (created_name[0] !== ".") {
-                    setFiles(f => [...f, { name: created_name, type: item.type, modified: date_modified }]);
+        const getFsList = () => {
+            if (channel.current !== null)
+                channel.current.close();
+
+            const currentPath = path.slice(0, pathIndex).join("/");
+            channel.current = cockpit.channel({
+                payload: "fslist1",
+                path: `/${currentPath}`,
+                superuser: "try",
+                watch: true,
+            });
+
+            setFiles([]);
+            channel.current.addEventListener("message", (ev, data) => {
+                const item = JSON.parse(data);
+                if (item.event === "present") {
+                    if (item.path[0] !== ".") {
+                        // TODO: Show hidden files if the user has enabled that option
+                        setFiles(f => [...f, { name: item.path, type: item.type, modified: item.modified, size: item.size, owner: item.owner, group: item.group }]);
```

Comment:
You must not call setFiles() for every single event here. Collect it in a local list, and then setFiles() once in the "ready" handler.

We already sorted all of this out in #34, but apparently this PR now contains an older version. Can you please sync this up?

Reply:
Note that https://github.com/KKoukiou/cockpit-navigator/pull/34/files#diff-73ba1bb45657b12f529b48ba4515bc802efe0eb6184f56c364c6f5c9753b7441R64 is wrong -- "ready" is too early, it needs to be the "done" handler.